salt/state.py

```diff
@@ -72,9 +72,13 @@
 HighData = dict[str, Union[Mapping[str, Any], list[Union[Mapping[str, Any], str]]]]
 LowChunk = dict[str, Any]
 
+K = TypeVar("K")
+V = TypeVar("V")
+Pair = tuple[K, V]
+HighDataStateArgsDef = Union[Mapping[K, V], Iterable[Union[Mapping[K, V], str]]]
```

Comment:
```suggestion
HighDataStateArgsDef = Mapping[K, V] | Iterable[Mapping[K, V] | str]
```

Minor suggestion: AFAIK 3008.0 is Python 3.10+ only, so you can use the `|` type operator. Unsure why `pyupgrade` does not catch this.